### ISSUE TITLE
Build broken on latest Rust nightly (27901849e 2015-03-25)

### ISSUE THREAD

**Issue by hawkw** [Marwes/parser-combinators#25]
```
/home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/parser-combinators-0.2.4/src/combinator.rs:156:36: 156:37 error: unable to infer enough type information about `_`; type annotations required [E0282]
/home/travis/.cargo/registry/src/github.com-1ecc6299db9ec823/parser-combinators-0.2.4/src/combinator.rs:156     NotFollowedBy(try(parser).then(f as fn (_) -> _)
                                                                                                                                               ^
error: aborting due to previous error
Build failed, waiting for other jobs to finish...
```

See [here](https://travis-ci.org/hawkw/seax#L133) for the whole build. I don't _think_ this one's my fault.

**Comment by Marwes** [comment]
Seems to be a regression (possibly intentional?) in how rust infers casts using `as`. Seems that adding type annotations instead should make it compile but I will investigate some more before uploading a fix in case I need to post on issue to rust as well.

**Comment by hawkw** [comment]
Great, thanks for checking it out.

**Comment by Marwes** [comment]
Fixed with 3a75699a56e0737e6169f468c53fa83972ca3451. Seems to be this issue in rustc https://github.com/rust-lang/rust/issues/23699
